### ISSUE TITLE
PHP API to find out if user has permissions to view a project by id?

### ISSUE THREAD

**Issue by sergedirect** [kanboard/kanboard#3029]
Can't find documentation for reusable functions?
e.g. I need to find out if logged in user has permissions to view project by ID, 
so I can generate an appropriate response from a plugin.

**Comment by sergedirect** [comment]
I have found the documentation for plugins.
I Still have few questions, though.
I want to create a plugin that will prevent users to see tasks that are not for them or if user has inappropriate user role or any another exception

I have found this documentation @
https://kanboard.net/documentation/plugin-authorization-architecture

What I need is to be able to cross check whether current, authenticated user has privileges to view task.
For this I need:

1. User data - is there global function or variable I can use to pull current user data as an object or an array.
e.g. getUser();

2. Current project, task, subtask data in an object or an array 
e.g. getTask() getProject() etc.

I am thinking of a very simple solution e.g.

if(getUser()['userCategory'] !== getTask['taskCategory'])
or
if(getUser()['userRole'] !== getTask['taskCategory'])
die(); 

etc..

P.S. Good coding practices & what a nice project.
Plugin API could be extended dramatically by providing easy to access data via Objects, global Functions

**Comment by fguillot** [comment]
You should take a look to the class `ProjectRoleHelper`. If some features are missing, you could send a pull-request.

**Comment by sergedirect** [comment]
Thank you @fguillot 
I've managed to access the user project role by:
$this->helper->projectRole->getProjectUserRole($project_id);

I was playing with KanBoard for over a year, took me a long time to get around the API, yet still getting everything but getting there slowly :)

My end goal was to show tasks that are assigned to a current user only if a user is not a project manager.

So now I have managed to achieve by tampering with BaseController and adding few exceptions in project and task view functions.

I believe expanding the API docs could help coding plugins in a more straightforward way.
Currently, I can only afford to do little tweaks and even this takes ages...

But hey @fguillot, thank you for this solid soft, with all the minuses, it is still my company's preferred PM tool.

Best Wishes
Sergei
